wagtail_streamfield_migration_toolkit/utils.py

```diff
@@ -1,6 +1,181 @@
+from collections import deque
+from wagtail.blocks import ListBlock, StreamBlock, StructBlock
 
+from wagtail_streamfield_migration_toolkit.operations import BaseBlockOperation
 
-def apply_changes_to_raw_data(raw_data, block_path, operation, *args, **kwargs):
-    altered_raw_data = raw_data
-    # Logic will be here 
-    return altered_raw_data
\ No newline at end of file
+
+# TODO complete descriptions
+# TODO see if a BlockPath Class will be helpful for operations involving moving blocks
+
+
+def is_block_in_path(block, block_path):
+    return block["type"] == block_path[0]
+
+
+def get_block_structure_type(block_def):
+    # for blocks with children, returns whether it is a StreamBlock, StructBlock or ListBlock
+
+    if isinstance(block_def, StreamBlock):
+        return StreamBlock
+    elif isinstance(block_def, StructBlock):
+        return StructBlock
+    elif isinstance(block_def, ListBlock):
+        return ListBlock
+
+    # TODO check if handling this is necessary
+    return None
+
+
+def parse_block_path(block_path_str):
+    # currently block_path_str should be a '.' separated list of names of the blocks from
+    # the top level block to the nested block which is subject to the operation
+    # eg:- `simplestream.char1`
+    block_path = block_path_str.split(".")
+    return block_path
+
+
+def get_altered_or_unchanged_block(
+    block,
+    block_def,
+    block_path,
+    operation: BaseBlockOperation,
+    is_list_child=False,
+    **kwargs
+):
+    """
+    TODO complete description
+    """
+
+    block_structure_type = get_block_structure_type(block_def)
```

Comment:
Suggestion: I would remove this, and just check the type of `block_def` in the clauses below, like `elif isinstance(block_def, ListBlock):`